File: admin-frontend/src/schemas/user-permission.ts
import { CreatedUpdatedAt } from './generics';
import { PermissionEmbedded } from './permission';
import { RoleEmbedded } from './role';

export interface UserPermissionCreate {
  id: string;
}

interface BaseUserPermission extends CreatedUpdatedAt {
  user_id: string;
  permission_id: string;
  from_role_id: string | null;
}

export interface UserPermission extends BaseUserPermission {
  permission: PermissionEmbedded;
  from_role: RoleEmbedded | null;
}

File: fief/schemas/user_permission.py
from pydantic import UUID4

from fief.schemas.generics import BaseModel, CreatedUpdatedAt
from fief.schemas.permission import PermissionEmbedded
from fief.schemas.role import RoleEmbedded


class UserPermissionCreate(BaseModel):
    id: UUID4


class BaseUserPermission(CreatedUpdatedAt):
    user_id: UUID4
    permission_id: UUID4
    from_role_id: UUID4 | None = None


class UserPermission(BaseUserPermission):
    permission: PermissionEmbedded
    from_role: RoleEmbedded | None
